crypto/serializer/serializer_pkey.c

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include <openssl/ui.h>
+#include <openssl/params.h>
+#include <openssl/serializer.h>
+#include <openssl/core_names.h>
+#include "internal/provider.h"
+#include "internal/property.h"
+#include "crypto/evp.h"
+#include "serializer_local.h"
+
+int OSSL_SERIALIZER_CTX_set_cipher_name(OSSL_SERIALIZER_CTX *ctx,
+                                        const char *cipher_name)
+{
+    OSSL_PARAM params[] = { OSSL_PARAM_END, OSSL_PARAM_END };
+
+    params[0] = OSSL_PARAM_construct_utf8_string(OSSL_SERIALIZER_PARAM_CIPHER,
```

Comment:
We don't seem to have support here for setting the properties (which must be set at the same time as the cipher name).

Reply:
Good point, will fix

Reply:
Done